packages/astro/src/content/mutable-data-store.ts

```diff
@@ -0,0 +1,366 @@
+import { promises as fs, type PathLike, existsSync } from 'node:fs';
+import * as devalue from 'devalue';
+import { imageSrcToImportId, importIdToSymbolName } from '../assets/utils/resolveImports.js';
+import { AstroError, AstroErrorData } from '../core/errors/index.js';
+import { type DataEntry, DataStore, type RenderedContent } from './data-store.js';
+import { contentModuleToId } from './utils.js';
+
+const SAVE_DEBOUNCE_MS = 500;
+
+export class MutableDataStore extends DataStore {
```

Comment:
Maybe we should call it `FileSystemDataStore`? Or if I misunderstood its purpose, maybe we could add a small JSDoc block that explains its purpose

Reply:
The base class doesn't have any write methods, so it's the fact that this one does (so is mutable) which is the important part, rather than the fact it uses the filesystem.

Reply:
I've added jsdocs to both